src/test/java/did/DIDDocumentTest.java

```diff
@@ -0,0 +1,171 @@
+        services.add(service);
+        return services;
+    }
+
+    private List<Authentication> createAuthentications(Map<String, Object> authenticationJsonLdObject) {
+        Authentication authentication = Authentication.build(authenticationJsonLdObject);
+        List<Authentication> authentications = new ArrayList<>();
+        authentications.add(authentication);
+        return authentications;
+    }
+
+    private List<PublicKey> createPublicKeys(Map<String, Object> publicKeyJsonLdObject) {
+        PublicKey publicKey = PublicKey.build(publicKeyJsonLdObject);
+        List<PublicKey> publicKeys = new ArrayList<>();
+        publicKeys.add(publicKey);
+        return publicKeys;
+    }
+
+    private Map<String, Object> createAnySimpleJsonLdObject(String id, String anyId) {
+        Map<String, Object> publicKeyJsonLdObject = new LinkedHashMap<>();
+        publicKeyJsonLdObject.put(id, anyId);
+        return publicKeyJsonLdObject;
+    }
+
+    @DisplayName("Raw DIDDocument returns null for JsonLdProperties")
+    @Test
+    void given_rawDIDDocument_when_gettersAreCalled_then_nullReturned() {
+        DIDDocument didDocument = DIDDocument.build();
+        assertNull(didDocument.getContexts());
+        assertNull(didDocument.getPublicKeys());
+        assertNull(didDocument.getId());
+        assertNull(didDocument.getAuthentications());
+        assertNull(didDocument.getServices());
+    }
+
+    @DisplayName("Context URI allowed")
+    @Test
+    void given_URIContext_when_getContextsCalled_then_contextIsReturned() throws URISyntaxException {
+
+        URI context = new URI("www.danubetech.com");
+        final String documentId = "documentId";
+        DIDDocument didDocument = DIDDocument.build(context, documentId, null,null,null);
+        assertEquals(context, didDocument.getContexts().get(0));
+    }
+
+    @DisplayName("Context List<?> allowed")
+    @Test
+    void given_AnyListContext_when_getContextsCalled_then_contextIsReturned() {
+        Object context = new ArrayList<>();
+        ((ArrayList<Integer>) context).add(123);
+        ((ArrayList<String>) context).add("Danubetech");
+        final String documentId = "documentId";
+        DIDDocument didDocument = DIDDocument.build(context, documentId, null,null,null);
+        assertEquals(didDocument.getContexts().get(0), ((ArrayList<String>) context).get(0));
+        assertEquals(didDocument.getContexts().get(1), ((ArrayList<String>) context).get(1));
+    }
+
+    @DisplayName("Context int return Null")
+    @Test
+    void given_intContext_when_getContextsCalled_then_returnNull() {
+        int context = 111111 ;
+        final String documentId = "documentId";
+        DIDDocument didDocument = DIDDocument.build(context, documentId, null,null,null);
+        assertNull(didDocument.getContexts());
+    }
+
+    @DisplayName("Null DocumentId and verify the Exception")
+    @Test
+    void given_Null_documentId_and_verify_exception(){
```

Comment:
```suggestion
     @DisplayName("Null DocumentId returns DidDocument skeleton")
    @Test
    void given_documentIdIsNull_when_getContextsCalled_then_DidDocumentSkeletonReturned(){
        DIDDocument didDocument = DIDDocument.build(null,null,null,null,null);
        assertNotNull(didDocument.getContexts());
        assertNull(didDocument.getId());
    }
```

Reply:
I ws re-checking the behaviour if the doumentId is null and in fact there is a DidDocument returned which has a context but the ID is null. So I extended and renamed the test.  

The reason I called it skeleton is, that there is a file called `diddocument-skeleton.jsonld` which is used to construct an empty DidDoc.